recipes/qt/all/conanfile.py

```diff
@@ -0,0 +1,685 @@
+            args += ['QMAKE_CXXFLAGS+="-ftemplate-depth=1024"']
+
+        if self.options.qtwebengine and self.settings.os == "Linux":
+            args += ['-qt-webengine-ffmpeg',
+                     '-system-webengine-opus']
+
+        if self.options.config:
+            args.append(str(self.options.config))
+
+        os.mkdir('build_folder')
+        with tools.chdir('build_folder'):
+            with tools.vcvars(self.settings) if self.settings.compiler == "Visual Studio" else tools.no_op():
+                build_env = {"MAKEFLAGS": "j%d" % tools.cpu_count(), "PKG_CONFIG_PATH": [self.build_folder]}
+                if self.settings.os == "Windows":
+                    build_env["PATH"] = [os.path.join(self.source_folder, "qt5", "gnuwin32", "bin")]
+                with tools.environment_append(build_env):
+                    self.run("%s/qt5/configure %s" % (self.source_folder, " ".join(args)), run_environment=True)
+                    if tools.os_info.is_macos:
+                        with open("bash_env", "w") as f:
+                            f.write('export DYLD_LIBRARY_PATH="%s"' % ":".join(RunEnvironment(self).vars["DYLD_LIBRARY_PATH"]))
+                    with tools.environment_append({
+                        "BASH_ENV": os.path.abspath("bash_env")
+                        }) if tools.os_info.is_macos else tools.no_op():
+                        self.run(self._make_program(), run_environment=True)
+
+    def package(self):
+        with tools.chdir('build_folder'):
+            self.run("%s install" % self._make_program())
+        with open(os.path.join(self.package_folder, "bin", "qt.conf"), 'w') as f:
+            f.write("""[Paths]
+Prefix = ..
+ArchData = bin/archdatadir
+HostData = bin/archdatadir
+Data = bin/datadir
+Sysconf = bin/sysconfdir
+LibraryExecutables = bin/archdatadir/bin
+Plugins = bin/archdatadir/plugins
+Imports = bin/archdatadir/imports
+Qml2Imports = bin/archdatadir/qml
+Translations = bin/datadir/translations
+Documentation = bin/datadir/doc
+Examples = bin/datadir/examples""")
+        self.copy("*LICENSE*", src="qt5/", dst="licenses")
+        for module in self._submodules:
+            if module != 'qtbase' and not getattr(self.options, module):
+                tools.rmdir(os.path.join(self.package_folder, "licenses", module))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        for file_to_remove in glob.glob(os.path.join(self.package_folder, "lib", "*.la*")):
+            os.remove(file_to_remove)
+        for file_to_remove in glob.glob(os.path.join(self.package_folder, "lib", "*.pdb*")):
+            os.remove(file_to_remove)
+        for root, dirs, files in os.walk(os.path.join(self.package_folder, "bin")):
+            for name in files:
+                if name.endswith('.pdb'):
+                    os.remove(os.path.join(root, name))
+            
```

Comment:
Nit: This line contains trailing whitespaces in the file. Not visible on github for unknown reasons.